signbank/video/tests/test_views.py

```diff
@@ -196,3 +197,116 @@ def test_upload_glossvideo_with_custom_redirect(self):
+            name="testdataset", signlanguage=self.signlanguage)
+        self.testgloss = Gloss.objects.create(
+            idgloss="testgloss", dataset=self.dataset, created_by=self.user, updated_by=self.user)
+        self.language_en = Language.objects.create(
+            name='English', language_code_2char='en', language_code_3char='eng')
+
+        testfile = SimpleUploadedFile(
+            'testvid.mp4', b'data \x00\x01', content_type='video/mp4')
+        self.glossvid = GlossVideo.objects.create(gloss=self.testgloss,
+                                            is_public=False,
+                                            dataset=self.testgloss.dataset,
+                                            videofile=testfile)
+
+    def test_export_csv_with_no_permission(self):
+        response = self.client_noperm.get(reverse('video:export_glossvideos_csv'))
+          # Make sure user does not have change_gloss permission.
+        self.assertFalse(response.wsgi_request.user.has_perm(
+            'dictionary.export_dictionary'))
+        # Should return 302 Found, or 403 Forbidden
+        self.assertTrue(response.status_code == 403 or 302)
+
+    def test_export_csv_with_public(self):
+        self.glossvid.is_public = True
+        self.glossvid.save()
+
+        response = self.client.get(reverse('video:export_glossvideos_csv'))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response['Content-Type'], 'text/csv')
+
+        csv_data = b''.join(response.streaming_content).decode('utf-8')
+        reader = csv.reader(csv_data.splitlines())
+        csv_rows = list(reader)
+
+        expected_glossvid_id = str(self.glossvid.pk)
+        self.assertIn(expected_glossvid_id, [row[0] for row in csv_rows])
+
+
+    def test_export_csv_with_not_public_no_filter(self):
+        response = self.client.get(reverse('video:export_glossvideos_csv'))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response['Content-Type'], 'text/csv')
+
+        csv_data = b''.join(response.streaming_content).decode('utf-8')
+        reader = csv.reader(csv_data.splitlines())
+        csv_rows = list(reader)
+
+        self.assertEqual(len(csv_rows), 1) # Header only
+
+
+    def test_export_csv_with_gloss_ids(self):
+        self.glossvid.is_public = True
+        self.glossvid.save()
+
+        response = self.client.get(reverse('video:export_glossvideos_csv'), { 'gloss': [str(self.testgloss.pk)] })
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response['Content-Type'], 'text/csv')
+
+        csv_data = b''.join(response.streaming_content).decode('utf-8')
+        reader = csv.reader(csv_data.splitlines())
+        csv_rows = list(reader)
+        expected_glossvid_id = str(self.glossvid.pk)
+        self.assertIn(expected_glossvid_id, [row[0] for row in csv_rows])
+
+        response = self.client.get(reverse('video:export_glossvideos_csv'), { 'gloss': [-1] })
```

Comment:
**suggestion** this looks like a separate test case - it's looking for the result when an invalid pk is provided